File: app/src/main/java/com/datarockets/mnchkn/fragments/players/PlayerInteractorImpl.java
package com.datarockets.mnchkn.fragments.players;

import android.content.Context;

import com.datarockets.mnchkn.MunchkinApplication;
import com.datarockets.mnchkn.activities.BaseInteractor;
import com.datarockets.mnchkn.store.ScoreService;

import javax.inject.Inject;

public class PlayerInteractorImpl extends BaseInteractor implements PlayerInteractor {

    @Inject
    ScoreService mScoreService;

    public PlayerInteractorImpl(Context context) {
        super(context);
    }

    @Override
    public void increaseLevelScore(int score, OnUpdatedListener listener) {
        listener.onUpdateLevel(mScoreService.increaseScore(score));
    }

    @Override
    public void decreaseLevelScore(int score, OnUpdatedListener listener) {
        listener.onUpdateLevel(mScoreService.decreaseScore(score));
    }

    @Override
    public void increaseStrengthScore(int score, OnUpdatedListener listener) {
        listener.onUpdateStrength(mScoreService.increaseScore(score));
    }

    @Override
    public void decreaseStrengthScore(int score, OnUpdatedListener listener) {
        listener.onUpdateStrength(mScoreService.decreaseScore(score));
    }

    @Override
    protected void setUpComponent(Context context) {
        MunchkinApplication.get(context).getAppComponent().inject(this);
    }
}

File: app/src/main/kotlin/com/datarockets/mnchkn/fragments/players/PlayerInteractorImpl.kt
package com.datarockets.mnchkn.fragments.players

import android.content.Context

import com.datarockets.mnchkn.MunchkinApplication
import com.datarockets.mnchkn.activities.BaseInteractor
import com.datarockets.mnchkn.store.ScoreService

import javax.inject.Inject

class PlayerInteractorImpl(context: Context) : BaseInteractor(context), PlayerInteractor {

    @Inject
    lateinit var mScoreService: ScoreService

    override fun increaseLevelScore(score: Int, listener: PlayerInteractor.OnUpdatedListener) {
        listener.onUpdateLevel(mScoreService.increaseScore(score))
    }

    override fun decreaseLevelScore(score: Int, listener: PlayerInteractor.OnUpdatedListener) {
        listener.onUpdateLevel(mScoreService.decreaseScore(score))
    }

    override fun increaseStrengthScore(score: Int, listener: PlayerInteractor.OnUpdatedListener) {
        listener.onUpdateStrength(mScoreService.increaseScore(score))
    }

    override fun decreaseStrengthScore(score: Int, listener: PlayerInteractor.OnUpdatedListener) {
        listener.onUpdateStrength(mScoreService.decreaseScore(score))
    }

    override fun setUpComponent(context: Context) {
        MunchkinApplication.get(context).appComponent.inject(this)
    }
}
